src/store/actions/contactsActions.js

```diff
@@ -1,87 +1,132 @@
+      }
+
+      // console.log(profilesArray);
+
+      profilesArray.forEach(profile => {
+        firestore
+          .collection("users")
+          .doc(profile.uid)
+          .update({
+            contacts: profile.contacts
+          });
       });
-      const contactArray = userInfo[0].contacts;
-      console.log('add action called', contactArray)
-      return contactArray
-    })
-    .then((contactArray) => {
-      contactArray.push(searchedEmail[0])
-      firestore.collection('users').doc(currentUserUid).update({
-      contacts: contactArray
-      })
     })
-    .then(() => 
+    .then(() =>
     {
       dispatch({ type: "ADD_CONTACT" }
       );
     }).catch((err) => {
       dispatch({ type: "ADD_CONTACT_ERROR", err})
     })
-
   };
 };
 
 export const deleteContact = (searchedEmail, currentUserUid) => {
   return (dispatch, getState, { getFirestore }) => {
-    const firestore = getFirestore()
-    const userID = getState().firebase.auth.uid // later for mapping userid to contacts
-    // onSnapshot - try to implement for auto-updating
-    firestore.collection('users').where('uid', '==', userID).get().then(querySnapshot => {
-      let userInfo = []
-      querySnapshot.forEach(doc => {
-        userInfo.push(doc.data())
-      });
-      const contactArray = userInfo[0].contacts;
-      return contactArray
-    })
-    .then((contactArray) => {
-      const filteredArray = contactArray.filter(contact => contact.email !== searchedEmail)
-      firestore.collection('users').doc(currentUserUid).update({
-      contacts: filteredArray
+    const firestore = getFirestore();
+    const userID = getState().firebase.auth.uid; // later for mapping userid to contacts
+    console.log(userID, searchedEmail);
+    Promise.all([
+      firestore
+        .collection("users")
+        .where("uid", "==", userID)
+        .get(),
+      firestore
+        .collection("users")
+        .where("uid", "==", searchedEmail.uid)
+        .get()
+    ])
+    .then(profiles => {
+      const profilesArray = [];
+      profiles.forEach(profile => {
+        profile.forEach(querySnapshot => {
+          const data = querySnapshot.data();
+          profilesArray.push(data);
+        });
       })
-      // return filteredArray
-      console.log('delete dispatch called', filteredArray)
-      dispatch({ type: "DELETE_CONTACT", filteredArray }
-      );
-    })
-    // .then((filteredArray) => 
-    // {
-    // })
-    .catch((err) => {
-      dispatch({ type: "DELETE_CONTACT_ERROR", err})
-    })
 
+      
+        const currentFilteredContects = profilesArray[0].contacts.filter(item => {return item.uid !== searchedEmail.uid})
+        firestore
+          .collection("users")
+          .doc(userID)
+          .update({
+            contacts: currentFilteredContects
+          })
+      
+        const oponentFilteredContects = profilesArray[1].contacts.filter(item => {return item.uid !== userID})
```

Comment:
"opponent" is generally a term used for somebody that is working against you.
Please "otherUserFilteredContacts" or something that is easier to understand.